crux_http/src/lib.rs

```diff
@@ -87,6 +87,26 @@ where
         });
     }
 
+    pub fn post<Res, F>(&self, url: Url, callback: F)
+    where
+        Res: serde::de::DeserializeOwned,
+        F: Fn(Res) -> Ev + Send + Clone + 'static,
+    {
+        let ctx = self.context.clone();
+        self.context.spawn(async move {
+            let request = HttpRequest {
+                method: HttpMethod::Post.to_string(),
+                url: url.to_string(),
+            };
+            let resp = ctx.request_from_shell(request).await;
+
+            let data = serde_json::from_slice::<Res>(&resp.body)
+                .expect("TODO: do something sensible here");
```

Comment:
Should `Output` be `Result<HttpResponse, HttpError>`? Http is (famously) fallible after all.

Reply:
yeah I agree. We should also look at @obmarg 's http capability based on surf. (so didn't want to jump to far just yet)